### PR TITLE
DataGrid(T1161941): Header Filter cannot switch options

### PR DESCRIPTION
**Pull request by williamvinogradov**
See the SC ticket: [T1161941](https://supportcenter.devexpress.com/ticket/details/T1161941)

### CODE REVIEW

**Review by const314 — DISMISSED** [review]

File: js/core/utils/common.js [hunk 1]
```diff
@@ -262,58 +262,60 @@ export const grep = function(elements, checkFunction, invert) {
...
-        if(!(propertyName in object1)) {
-            return false;
-        }
+    if(keys1.length !== keys2.length) {
+        return false;
     }
 
-    return true;
+    return !keys1.some((key) =>
+        !Object.prototype.hasOwnProperty.call(object2, key)
```
Comment (pomahtri) [inline js/core/utils/common.js]: According to [doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys), `Object.keys` already returns only own properties, so we can remove `hasOwnProperty`

File: js/core/utils/common.js [hunk 2]
```diff
@@ -262,58 +262,59 @@ export const grep = function(elements, checkFunction, invert) {
...
-    for(const propertyName in object2) {
-        if(!(propertyName in object1)) {
-            return false;
-        }
+    if(keys1.length !== keys2.length) {
+        return false;
     }
 
-    return true;
+    const keys2Set = keys2.reduce((set, key) => set.add(key), new Set());
```
Comment (pomahtri) [inline js/core/utils/common.js:284]:
```suggestion
    const keys2Set =new Set(keys2);
```

Comment (pomahtri) [inline js/core/utils/common.js]:
```suggestion
    const keys2Set = new Set(keys2);
```

File: js/core/utils/common.js (hunk 1 above)
Comment (williamvinogradov) [reply js/core/utils/common.js]: Yup, I agree with you
Let's remove it

**Review by pomahtri — DISMISSED** [review]

**Review by pomahtri — APPROVED** [review]

**Review by sergepilipchuk — APPROVED** [review]

**Review by const314 — APPROVED** [review]

**Pull request merged**